packages/vda-node-client/src/getNodes.ts

```diff
@@ -0,0 +1,148 @@
+    return await executeFunction(
+        network,
+        'isRegisteredNodeEndpoint',
+        'Failed to check node endpoint',
+        endpointUri
+    );
+}
+
+/**
+ * Get a node by name
+ * @param network One of Verida supported network
+ * @param name Node name
+ * @returns A storage node if name is registered. Otherwise rejected
+ */
+export async function getNodeByName(network: string, name: string) {
+    return await executeFunction(
+        network,
+        'getNodeByName',
+        'Failed to get node by name',
+        name
+    );
+}
+
+/**
+ * Get a node by DID address
+ * @param network One of Verida supported network
+ * @param didAddress Node address
+ * @returns A storage node if address is registered. Otherwise rejected
+ */
+export async function getNodeByAddress(network: string, didAddress: string) {
+    return await executeFunction(
+        network,
+        'getNodeByAddress',
+        'Failed to get node by address',
+        didAddress
+    );
+}
+
+/**
+ * Get a node by endpointUri
+ * @param network One of Verida supported network
+ * @param endpointUri EndpointUri of a storage node
+ * @returns A storage node if endpointUri is registered. Otherwise rejected
+ */
+export async function getNodeByEndpoint(network: string, endpointUri: string) {
+    return await executeFunction(
+        network,
+        'getNodeByEndpoint',
+        'Failed to get node by endpoint',
+        endpointUri
+    );
+}
+
+/**
+ * Get nodes by country and status
+ * @param network Target network
+ * @param countryCode Country code of nodes
+ * @param status Status of nodes
+ */
+export async function getNodesByCountry(
+    network: string, 
+    countryCode: string, 
+    status?: EnumStatus) {
+    if (status === undefined) {
+        return await executeFunction(
+            network,
+            'getNodesByCountry',
+            'Failed to get nodes by country',
+            countryCode
+        );
+    } else {
+        return await executeFunction(
+            network,
+            'getNodesByCountryAndStatus',
+            'Failed to get nodes by country and status',
+            countryCode,
+            status
+        );
+    }
+}
+
+/**
+ * Get nodes by region and status
+ * @param network Target network
+ * @param regionCode Region code of nodes
+ * @param status Status of nodes
+ */
+export async function getNodesByRegion(
+    network: string, 
+    regionCode: string, 
+    status?: EnumStatus) {
+    if (status === undefined) {
+        return await executeFunction(
+            network,
+            'getNodesByRegion',
+            'Failed to get nodes by region',
+            regionCode
+        );
+    } else {
+        return await executeFunction(
+            network,
+            'getNodesByRegionAndStatus',
+            'Failed to get nodes by region and status',
+            regionCode,
+            status
+        );
+    }
+}
```

Comment:
Can we have a `getNodes()` that returns all the nodes?

That will be helpful for our node explorer

Reply:
Added `getNodesByStatus()` function.